src/ng/directive/input.js

```diff
@@ -1152,10 +1158,18 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     }
   };
 
+  function ieListener(ev) {
+    var val = element.val();
+    if (val === oldVal && !viewValueUpdated) return;
```

Comment:
Is it possible that the validity changed but the value did not (similar to the `value === '' && ctrl.$$hasNativeValidators` check in `listener`)? Although I'm not seeing how that would have been handled previously either (in the `!hasEvent('input')` case)...